day2/gift.py

```diff
@@ -0,0 +1,42 @@
+class Gift:
+    def __init__(self):
+        self.total = 0
+
+    def calculation_of_wrapping_paper(self, dimensions):
+        dimensions_new = remove_spaces_white(dimensions)
+
+        for dimension in dimensions_new:
+            result = []
+            length, width, height = list(map(int, dimension.split('x')))
+
+            result.append(length * width)
+            result.append(width * height)
+            result.append(height * length)
+            result_min = min(result)
+
+            mult_result = lambda result: [value * 2 for value in result]
+            result = mult_result(result)
+            self.total += sum(result) + result_min
+
+        return self.total
+
+    def calculation_of_feet_ribbon(self, dimensions):
+        dimensions_new = remove_spaces_white(dimensions)
+
+        for dimension in dimensions_new:
+            result = []
+            length, width, height = list(map(int, dimension.split('x')))
+
+            result.append(length)
+            result.append(width)
+            result.append(height)
+            result_sort = sorted(result)
+
+            result_sum = result_sort[0] * 2 + result_sort[1] * 2
+            result_mul = result[0] * result[1] * result[2]
+            self.total += result_mul + result_sum
```

Comment:
Tudo isso me pareceu meio confuso! Esses `append` com `sorted` e a porra toda. E quando eu tentei entender o pq disso, eu fiquei mais confuso ainda. Então eu abri o meu advent-of-code pra ver como eu tinha feito, e vi que pareceu bem mais simples.
Não sei se iria funciona também pra Python, mas acredito que sim.
Talvez seja algo como:

```python
ribbon = 0
for dimension in dimensions_new:
    length, width, height = list(map(int, dimension.split('x')))
    min = [length+width, width+height, height+length].min * 2
    ribbon += min + length*width*height

return ribbon
```

Reply:
@ricardoruwer não vejo confusão na lógica mas os nomes das variaveis não estão agradaveis mas vou fazer uma melhoria nesse código, não se se é possivel fazer como o exemplo que você deixou

Reply:
Mesmo que não tivesse "confusão" na lógica, parece que tem coisas a mais que precisava ter.
Tipo esses `append` e `sorted`, não sei pq tem e o que ele faz... Acho que não precisa disso...